vulcanai2/models/basenetwork.py

```diff
@@ -215,22 +219,22 @@ def criter_spec(self, value):
 
     def get_flattened_size(self, network):
         """
-        Returns the flattened output size of the conv network's last layer.
+        Returns the flattened output size of a Single Input ConvNet's last layer.
         :param network: The network to flatten
         :return: The flattened output size of the conv network's last layer.
         """
         with torch.no_grad():
-            x = torch.ones(1, *self.in_dim)
+            x = torch.ones(1, *self.in_dim[0])
```

Comment:
there should be no index-based retrievals. what if the conv wasn't in the first slot?

Reply:
I think I mentioned in the documentation "works only for Single Input case"

Reply:
why not make it work for all input cases?

Reply:
I wanted it to be a helper function to get the flattened size off a single input network, I think for multpple case, we can use get_output_shapes?

Reply:
we shouldn't have a conv specific function in basenetwork